riot/riot.mli

```diff
@@ -298,6 +298,13 @@ val shutdown : ?status:int -> unit -> unit
 val run : ?rnd:Random.State.t -> ?workers:int -> (unit -> unit) -> unit
 (** Start the Riot runtime using function [main] to boot the system *)
 
+val run_with_status : ?rnd:Random.State.t -> ?workers:int -> ?on_error:(([> `Msg of string ] as 'a) -> int) -> (unit -> (int, 'a) result) -> unit
```

Comment:
**suggestino**. If we are more general about the `'error` that we return and how to handle it, we can have a smaller error type for our `default_on_error` function.


```suggestion
val run_with_status : ?rnd:Random.State.t -> ?workers:int -> ?on_error:('error -> int) -> (unit -> (int, 'error) result) -> unit
```

This change should mean that you can return whatever error type you want, as long as you use an `on_error` function that supports it. If you use the default, then the only error type you can return is ` [| `Msg of string ] `.

Reply:
Oh I see, thanks :D

Reply:
I'm still getting [ `Msg of string ] not captible with 'error in the mli file unfortunately